src/lib.rs

```diff
@@ -237,3 +239,11 @@ impl<'s> CowUtils<'s> for &'s str {
         }
     }
 }
+
+fn changes_when_lowercased(c: char) -> bool {
+    !core::iter::once(c).eq(c.to_lowercase())
```

Comment:
Why the indirection with single-item iterator instead of just `c == c.to_lowercase()`?